Fix 1353: tab navigation via Options menu

Krismix1 opened the pull request:
Fix #1353 when attempting to change tabs via the Option menu. The description of `ParentView` remotely matches the behaviour needed, but perhaps it was initially intended to be used differently.

Open for feedback.

jesseduffield commented on pkg/gui/view_helpers.go:
```diff
@@ -392,3 +392,17 @@ func (gui *Gui) pageDelta(view *gocui.View) int {
 	return delta
 }
+
+func getTabbedView(gui *Gui) *gocui.View {
```
I added the ParentView field to my gocui fork a while back, before I introduced the concept of Contexts. I haven't officially deprecated it but it is in essence deprecated in favour of Contexts. We have a Context stack that typically starts at the bottom with the current side view and in this case ends at the top with our menu view. The `gui.currentStaticContext()` function starts at the top of the stack and goes down until it finds a static (i.e. non-popup) context.

I reckon we should do the following:
```go
func getTabbedView(gui *Gui) *gocui.View {
	// assumes only static contexts have tabs which is an assumption I consider safe
	context := gui.currentStaticContext()
	view, _ := gui.g.View(context.GetViewName())
	return view
}
```

Krismix1 replied:
Updated to static contexts, and removed the `ParentView` assignment to not add code using a deprecated field.

jesseduffield reviewed (changes requested):
Looking good! Also great work on the integration test. I've taken the liberty of pushing a slight tweak to it so that it's less brittle: we're now searching for the item by name in the menu rather than moving our cursor all the way to it, so that if we add more items to the menu down the line, the test won't break.

jesseduffield commented:
Looks like you'll also need to rebase onto the master branch to fix that linting issue

Krismix1 commented on pkg/gui/menu_panel.go:
```diff
@@ -74,11 +74,13 @@ func (gui *Gui) createMenu(title string, items []*menuItem, createMenuOptions cr
 	menuView.Title = title
 	menuView.FgColor = theme.GocuiDefaultTextColor
 	menuView.ContainsList = true
+	menuView.ParentView = parentView
```
Updated to static contexts, and removed the `ParentView` assignment to not add code using a deprecated field.

Krismix1 commented:
I have updated the changes to use the static contexts. Good idea with using the search instead of hardcoded navigation.
I have rebased the branch, let's this pipeline be green.

jesseduffield commented:
nice one :)
